src/btree/bt_sync.c

```diff
@@ -109,44 +109,53 @@ __sync_ref_is_obsolete(WT_SESSION_IMPL *session, WT_REF *ref)
 {
     WT_ADDR *addr;
     WT_CELL_UNPACK vpack;
+    WT_PAGE_MODIFY *mod;
 
     /* Ignore root pages as they can never be deleted. */
     if (__wt_ref_is_root(ref))
         return (false);
 
-    /* Guard against marking an in-memory page to be deleted. */
-    if (ref->state != WT_REF_DISK)
-        return (false);
-
     /* Ignore internal pages, these are taken care of during reconciliation. */
     if (!__wt_ref_is_leaf(session, ref))
         return (false);
 
+    /* In-memory ref address must be from an off page address. */
```

Comment:
@keithbostic Is my understanding correct that an in-memory page that `rec_result` is not empty must have an address from off-page?

Reply:
> Is my understanding correct that an in-memory page that rec_result is not empty must have an address from off-page?

No; we create pages  in memory in some cases (as one example, an insert split will create a new page in memory that was never read from disk). Those pages are reconciled normally, and so `rec_result` can have a value.

In short, `rec_result` tells you the result of the last reconciliation, but nothing else.

Reply:
Thanks. I updated the comments.